lib/oli_web/account_lookup_cache.ex

```diff
@@ -0,0 +1,134 @@
+defmodule OliWeb.AccountLookupCache do
+  @moduledoc """
+    Provides a cache that can be used for account lookups. This cache is backed by
+    Cachex for local storage and PubSub for remote distribution. Keys are set to expire
+    after 1 day in order to prevent stale data in our cache over a long time period.
+  """
+
+  use GenServer
+
+  alias Phoenix.PubSub
+
+  @cache_name :cache_account_lookup
+
+  # ----------------
+  # Client
+
+  def start_link(init_args),
+    do: GenServer.start_link(__MODULE__, init_args, name: __MODULE__)
+
+  def get(key),
+    do: GenServer.call(__MODULE__, {:get, key})
+
+  def delete(key),
+    do: GenServer.call(__MODULE__, {:delete, key})
+
+  def put(key, value),
+    do: GenServer.call(__MODULE__, {:put, key, value})
+
+  def update(key, value),
+    do: GenServer.call(__MODULE__, {:update, key, value})
+
+  # ----------------
+  # Server callbacks
+
+  def init(_) do
+    {:ok, _pid} = Cachex.start_link(@cache_name, stats: true)
+    PubSub.subscribe(Oli.PubSub, cache_topic())
+
+    {:ok, [], {:continue, :init}}
+  end
+
+  def handle_continue(:init, state) do
+    nodes = Node.list()
+
+    if length(nodes) > 0 do
+      # just pick a random node in the cluster to request the dump
+      send({__MODULE__, Enum.random(nodes)}, {:request_dump, self()})
```

Comment:
Is there a reason why it needs to be a random node and not just the first one?

Reply:
Nop, we can choose always the first one if we like it more. 

Tbh I don't know if `Node.list()` always return in the same order but if that the case, a random one could be better to not load the same node always, wdyt?

Reply:
Yeah, I agree 100%